server/asset/btc/btc.go

```diff
@@ -239,15 +240,38 @@ func (btc *Backend) Redemption(redemptionID, contractID []byte) (asset.Coin, err
 }
 
 // FundingCoin is an unspent output.
-func (btc *Backend) FundingCoin(coinID []byte, redeemScript []byte) (asset.FundingCoin, error) {
+func (btc *Backend) FundingCoin(ctx context.Context, coinID []byte, redeemScript []byte) (asset.FundingCoin, error) {
 	txHash, vout, err := decodeCoinID(coinID)
 	if err != nil {
 		return nil, fmt.Errorf("error decoding coin ID %x: %w", coinID, err)
 	}
-	utxo, err := btc.utxo(txHash, vout, redeemScript)
+
+	// Maybe don't do this, just a test:
+	type utxoRes struct {
+		*UTXO
+		error
+	}
+	res := make(chan *utxoRes)
+	go func() {
+		utxo, err := btc.utxo(txHash, vout, redeemScript)
+		select {
+		case res <- &utxoRes{utxo, err}:
+		case <-ctx.Done(): // discard late results
+		}
+	}()
+
+	var utxo *UTXO
+	select {
+	case <-ctx.Done():
+		err = asset.ErrRequestTimeout
+	case r := <-res:
+		utxo, err = r.UTXO, r.error
+	}
+
```

Comment:
This may be pure madness, since the rpc calls aren't actually canceled, just left to rot.  It's probably better not to do this and instead ignore the input arg, naming it `_ context.Context` to make it clear there's no actual timeout or cancellation possible with the BTC backend

Reply:
I've removed these acrobatics.  We can plan to use *decred's* rpcclient for BTC with all requests going through `RawRequest`.